packages/astro/src/cli/index.ts

```diff
@@ -133,6 +96,16 @@ export async function cli(args: string[]) {
 		logging.level = 'silent';
 	}
 
+	if (cmd === 'add') {
+		try {
+			const packages = flags._.slice(3) as string[];
+			await add(packages, { projectRoot, flags, logging });
```

Comment:
Following up to a different thread: `projectRoot` here is either the `--project-root` value or undefined if no flag given. This is different from `astroConfig.projectRoot`. 

You mentioned that moving this to run after `loadConfig` with the other commands would require too big of a refactor. Can you add a comment here explaining why this command runs so early, for future refactor-ers? 

My main hesitation here is that calling this `projectRoot` makes it sound like the official config value, but its operating very different here. I actually think there's a bug in your code if you use `--project-root`, since flags.projectRoot should be a string but you have typed the `add()` argument as a URL. I could be wrong here, but I don't think so.

My suggestion would be to make this change to make it clear that this is a non-standard idea of cwd, and not an officially resolve projectRoot.

```suggestion
            const cwd = projectRoot ? path.resolve(projectRoot) : process.cwd();
			await add(packages, { cwd, flags, logging });
```

Reply:
Thank you, great callout. Will update!

Reply:
Updated to use `cwd` and pass through to use the behavior defined in the `core/config.ts` file.